core/src/main/java/hudson/model/Executor.java

```diff
@@ -941,12 +941,7 @@ public static Executor of(Executable executable) {
             return null;
         }
         for (Computer computer : jenkins.getComputers()) {
-            for (Executor executor : computer.getExecutors()) {
-                if (executor.getCurrentExecutable() == executable) {
-                    return executor;
-                }
-            }
-            for (Executor executor : computer.getOneOffExecutors()) {
+            for (Executor executor : computer.getAllExecutors()) {
```

Comment:
I think you are arguing for more complex code on the grounds of performance in the absence of a proven performance degradation. That smells like premature optimization